tests/lib/rules/jsx-one-element-per-line.js

```diff
@@ -101,6 +101,20 @@ ruleTester.run('jsx-max-elements-per-line', rule, {
     parserOptions: parserOptions
   }, {
     code: [
+      '<div>',
+      '  <span />foo<input />',
+      '</div>'
+    ].join('\n'),
+    output: [
+      '<div>',
+      '  <span />foo',
```

Comment:
If the lack of whitespace before "foo" is important, then it's just as important after "foo" - meaning, this example can't be autofixed.

However,
```jsx
<div>
  <span /> foo <input />
</div>
```
definitely should become:
```jsx
<div>
  <span />
  foo
  <input />
</div>
```

Reply:
@ljharb I believe white space in jsx is differnt from HTML and in your example the two jsx snippets are actually different. Foo in the first snippet will be rendered with a whitspace on either side, while foo in the second snippet will have no white space. 

However even if I'm correct the rule still does need to be patched. Currently the autofix does change the white space under certain conditions which is obviously unacceptable.

Reply:
Using curlies and transforming into a string literal with spacing on either side seems the most accurate. Is this acceptible?

```jsx
<div>
  <span />
  {' foo '}
  <input />
</div>
```

Reply:
Sure, that'd be great!

As long as the autofix is not changing the resulting whitespace that's rendered, we're good.